src/base/misc/disks.c

```diff
@@ -1349,18 +1349,26 @@ int disk_is_bootable(const struct disk *dp)
 int disk_validate_boot_part(struct disk *dp)
 {
   int hdtype;
+
   if (dp->type != DIR_TYPE || dp->floppy)
     return 1;
+
   hdtype = fatfs_get_part_type(dp->fatfs);
   if (hdtype == -1)
     return 0;
-  if (!hdtype || hdtype == dp->hdtype)
+  if (!hdtype)
     return 1;
-  d_printf("DISK: changing hdtype from %i to %i\n", dp->hdtype, hdtype);
-  dp->hdtype = hdtype;
-  dp->sectors = -1;
+
+  if (dp->hdtype == 0 || dp->hdtype > hdtype) {
```

Comment:
Comparing `dp->hdtype > hdtype` looks
nasty, maybe only force hdtype if user
didn't specify anything? After all we don't
know what DOS versions can exist in the
wild, so if the user wants to test this particular
hdtype, he should be able to do that. We
normally should not overrule user's choices,
just if the things are automatic.

Reply:
Sure that would be my preference too, I just figured you wanted to keep the ability to fix the user's mistake. I'll fix it now.